src/foram3d/foram.ts

```diff
@@ -76,38 +80,66 @@ module Foram3D {
       newChamber.aperture = newAperture;
       newChamber.setAncestor(this.currentChamber);
 
+      this.prevChambers[2] = this.prevChambers[1];
+      this.prevChambers[1] = this.prevChambers[0];
+      this.prevChambers[0] = newChamber;
+
       return newChamber;
     }
 
     private calculateNewCenter(): THREE.Vector3 {
-      var currentOrigin, currentAperture, growthVector, horizontalRotationAxis,
-          verticalRotationAxis, newCenter;
-
-      currentOrigin = this.currentChamber.origin;
-      currentAperture = this.currentChamber.aperture;
+      var referenceLine, deviationSurfaceSpanning, phiDeviationAxis, growthVector,
+          newCenter;
 
-      // calculate initial growth vector (reference line)
+      referenceLine = new THREE.Vector3();
 
-      growthVector = new THREE.Vector3();
-      growthVector.subVectors(currentAperture, currentOrigin);
+      if (this.chambers.length == 1) {
+        referenceLine.subVectors(
+          this.prevChambers[0].aperture,
+          this.prevChambers[0].center
+        );
+      } else {
+        referenceLine.subVectors(
+          this.prevChambers[0].aperture,
+          this.prevChambers[1].aperture
+        );
+      }
 
-      // deviate growth vector from reference line
+      deviationSurfaceSpanning = new THREE.Vector3();
```

Comment:
Spodziewałbym się, że zmienna która ma w nazwie "Surface" byłaby wektorem normalnym jakiejś powierzchni. Z tego co widzę, tutaj tak nie jest.

To jest wektor pomiędzy ujściem poprzedniej i jeszcze wcześniejszej komory. Dobrze rozumiem, że razem z wektorem `referenceLine` tworzą płaszczyznę nazywaną w [tym](https://9c57cd9cc879c922da5c8615c74e9a4a7838e715.googledrive.com/secure/AH8CU7FSUZh2f0IdmoWSsTo2VXZLrrbUklkbBn8iPTDZ1awX4PIPNuhAJxcw126D5xhvnDLWdv4hU9hpHeWjnvvAJ4qOis7HvrBBAfFGmpxFU460QU-H-YTMLZKIKVdQyjyQOI-19CuPn7GntVxzDeSz-LEoLx7n6w2LjypWTaKz2i73yB7GGTHvkI8zqlkUv2gKtTr4HT6XIAOKM6zw34iDNlnxVwI63NbJnecezqRaTNP2bTUtLFweJuqvgDhm5x2ARQfKed2jEY0uVWa-73eolZl2_9KN12jqDc6U3EFuZvrUWupOWcSftj51c21PPAk_e4wpK02VRpXqjMpKHtM_Zc17O7BlZCUXeycjntuWU0GPnnZl0WQXukMTCjX4QUUbY99Yx2L2tb6ZyhDKK-ZT1kfhjY966ylJWOcvVauz4dSq_Gt1hONboBhaty9z6v6rcFgWoAj6NAzcXc7Vrs2cFgbG_-tgTkd0fhFbXgiq5J-yYXiEjm8ZkYpWQXVQ2_2V5kNTUC_KS8DO2Baapd3LujjPs7CwnIHproO85npUTc6dV7QDZq15R81i0KI-nw7yS81Sn8Yf/host/0B3_8swGBUJzrcGtpcThfeWhySlU#przebieg-symulacji) dokumencie **P**?